### PR TITLE
RHPAM-2696: Notification UI produces syntactically incorrect iso expressions:

### PR DESCRIPTION
**Pull request by handreyrc**
@romartin , @inodeman : Could you review this PR please?

### CODE REVIEW

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins retest please

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/widgets/TimeZonePickerView.java [hunk 1]
```diff
@@ -27,6 +27,10 @@
 
     String getValue();
 
+    void setValue(String value);
+
+    void setLocalTimeZone();
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/widgets/TimeZonePickerView.java:32]: Not sure yet if makes sense adding this `setLocalTimeZone` method, at least for the view class. I understand the presenter should be responsible for obtaining the "local" timezone, and then just passing the raw value (maybe an string) to the view. 

So maybe only adding the new `setValue` method above on the view class is godo enough to achieve same behavior, but making the presenter obtain the "local" value and passing the value to it. 

PS: Also, what means "local", from the browser?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/widgets/TimeZonePicker.java [hunk 2]
```diff
@@ -77,6 +77,13 @@ public void setValue(String value) {
     public void setValue(String value, boolean fireEvents) {
         String oldValue = current;
         current = value;
+
+        if (current.equals("0")) {
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/widgets/TimeZonePicker.java:81]: why this?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/widgets/TimeZonePickerViewImpl.java [hunk 3]
```diff
@@ -43,8 +43,10 @@
 
     private final double defaultOffset = new Date().getTimezoneOffset() * -1;
 
+    private String localTimeZone;
+
     @DataField
-    private Select tzSelect = new Select();
+    protected Select tzSelect = new Select();
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/widgets/TimeZonePickerViewImpl.java:49]: if you changed from `private` to `protected`, just for making this field visible to the unit tests, maybe it's better to just make it `package protected` (no access modifier), which results more restrictive.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/forms/widgets/TimeZonePickerTest.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    @Mock
+    private Select tzSelect;
+
+    private TimeZonePickerViewImpl view;
+
+    private TimeZonePicker tested;
+
+    @Before
+    public void setUp() {
+        GwtMockito.initMocks(this);
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/forms/widgets/TimeZonePickerTest.java:47]: Not really sure if this is necessary, as far as I remember I think once declaring some `Runner` class above (on the class declaration) it already implies initializing the mocks from the class itself.

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/widgets/TimeZonePickerView.java (hunk 1 above)
Comment (handreyrc) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/widgets/TimeZonePickerView.java]: @romartin Good catch! I moved the logic above to the view and it is now encapsulated in the setValue() method. I also removed the setLocalTimeZone method once it is not needed anymore. It looks better now.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/widgets/TimeZonePicker.java (hunk 2 above)
Comment (handreyrc) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/widgets/TimeZonePicker.java]: @romartin The timezone is set in a modal that is reused for adding new notifications or edit them as well, also it is not being created and destroyed as you add/edit an item, it is being hidden from the user and the values reset to default values defined in the code.  
In order to know if I should set the userTimezone(New Item) or set a value from an existing item in the list (Edit) I'm testing the value "0" that is already being set for new items.
WDYT?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/widgets/TimeZonePickerViewImpl.java (hunk 3 above)
Comment (handreyrc) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/widgets/TimeZonePickerViewImpl.java]: @romartin Fixed!

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/forms/widgets/TimeZonePickerTest.java (hunk 4 above)
Comment (handreyrc) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/forms/widgets/TimeZonePickerTest.java]: @romartin you are right It is not necessary, I removed it.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/widgets/TimeZonePicker.java (hunk 2 above)
Comment (romartin) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/widgets/TimeZonePicker.java]: ok, in some way it's necessary to have some flag to check ,so that's fine comparing the value to `0` for me. But it results a bit hard to follow on the first look, maybe just replacing the String `"0"` to some static final String field in the class, which name gives some clue about why this `"0"` is used for, eg: `VALUE_EMPTY = "0"`. WDYT? Well just trying to make this more readable, maybe adding a comment in that line should work too.

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins retest please

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/widgets/TimeZonePicker.java (hunk 2 above)
Comment (handreyrc) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/widgets/TimeZonePicker.java]: @romartin I agree, I've created a static field to make it easier to follow.  Thanks.

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins retest please

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/widgets/TimeZonePicker.java (hunk 2 above)
Comment (romartin) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/widgets/TimeZonePicker.java]: I just understood the condition now at the first read, great!! thanks a lot for this small things, I think become important at long term... :+1:

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins retest please

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins retest please

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins retest please

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by LuboTerifaj** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
jenkins execute full downstream production build

**Comment by romartin** [comment]
Please @handreyrc 

Keep an eye on this PR and try to get the artifacts ready from testing, from any of the builds. Once artifacts ready please ping @LuboTerifaj to start the review.

I've just fired the production build, see comment above

**Comment by handreyrc** [comment]
jenkins execute full downstream production build

**Comment by LuboTerifaj** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Hey @LuboTerifaj, Thanks for the information! I've just rebased to clean the previous failed builds and I triggered the FDB again.

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
please @LuboTerifaj missing review here.

**Comment by LuboTerifaj** [comment]
Hi @romartin missing FDB here.

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Please @handreyrc try to upload the artifacts somewhere in case Jenkisn is not generiting the artifacts, so @LuboTerifaj  can start testing, we're really close to the due date for this. And please keep both in touch tomorrow for moving this stuff forwrad. Thanks a lot guys!

**Comment by handreyrc** [comment]
@romartin, @LuboTerifaj : I've generated the artifacts locally and they are available through the link:
https://drive.google.com/file/d/1-jTDVkigGjRod-mGHRfrmUryYqhID7iO/view?usp=sharing
Please, let me know if it suits you. Thanks

**Comment by domhanak** [comment]
@handreyrc I used WAR from FDB for testing. I was able to:
1. Import - https://github.com/mfpc/bugtimer
2. Build it and Deploy it.
3. Start Process instance and the instance completed without issues
4. Created a process with Human task and added notification `date/time` 
5. Set the timezone to GMT-6
6. Closed properties panel and deselected task
7. Re selected task a verified the notification is unchanged (stil lshows GMT-6)
8. Did the same but Saved process and reopened it

LGTM

**Comment by domhanak** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Jenkins execute full downstream build
